src/day14.scala

```diff
@@ -0,0 +1,112 @@
+// using scala 3.0.2
+
+package day14
+
+import scala.collection.mutable
+import scala.util.Using
+import scala.io.Source
+
+@main def part1(): Unit =
+  val answer = part1(readInput())
+  println(s"The answer is: $answer")
+
+@main def part2(): Unit =
+  val answer = part2(readInput())
+  println(s"The answer is:\n$answer")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day14"))(_.mkString)
+
+type Polymer = List[Char]
+type CharPair = (Char, Char)
+type InsertionRules = Map[(Char, Char), Char]
+type Frequencies = Map[Char, Long]
+
+def part1(input: String): Long =
+  val (initialPolymer, insertionRules) = parseInput(input)
+  val finalPolymer = (0 until 10).foldLeft(initialPolymer)((polymer, _) => applyRule(polymer, insertionRules))
+  val frequencies: Frequencies = finalPolymer.groupMapReduce(identity)(_ => 1L)(_ + _)
+  val max = frequencies.maxBy((char, count) => count)._2
```

Comment:
Here too, you could use the “index-like” access to the pair:

```suggestion
  val max = frequencies.maxBy((char, count) => count)(1)
```

There are other places like that below.